app/models/webhook.rb

```diff
@@ -130,6 +130,9 @@ def render_source(source, event_name, payload)
   end
 
   def rendered_payload(event_name, payload)
+    if webhook_template == nil
+      raise ::Foreman::Exception.new(N_("Webhook template not found while firing webhook '%s'. Possible organization/location mismatch in the webhook template?"), "#{name}")
```

Comment:
After another look I agree with @adamruzicka that we'd could simply check for `nil` here instead of `rescue`ing. But I still agree that we need to `raise` here.

Also, there is a small nit to the exception itself: you don't need to wrap the name into interpolation, you can simply pass it to the initializator of the exception. More info about that is here: https://projects.theforeman.org/projects/foreman/wiki/Translating#Exceptions

```suggestion
      raise ::Foreman::Exception.new(N_("Webhook template not found while firing webhook '%s'. Possible organization/location mismatch in the webhook template?"), name)
```

Also, a :policeman: suggests: prefer `if modifier`, i.e. short-cut in Ruby for simpler `if`s: `something if condition`.

To ask :policeman: explicitly, you can run `bundle exec rubocop .` within the directory or you might use `bundle exec rubocop ../foreman_webhook` within Foreman's directory (haven't tried though, always run within current file tree).